gluten-core/src/main/scala/io/glutenproject/execution/BatchScanExecTransformer.scala

```diff
@@ -33,10 +33,17 @@ import org.apache.spark.sql.vectorized.ColumnarBatch
 
 import java.util.Objects
 
+/**
+ * Columnar Based BatchScanExec. Although keyGroupedPartitioning is not used, it cannot be deleted,
+ * it can make BatchScanExecTransformer contain a constructor with the same parameters as
+ * Spark-3.3's BatchScanExec. Otherwise, the corresponding constructor will not be found when
+ * calling TreeNode.makeCopy and will fail to copy this node during transformation.
+ */
 class BatchScanExecTransformer(
     output: Seq[AttributeReference],
     @transient scan: Scan,
-    runtimeFilters: Seq[Expression])
+    runtimeFilters: Seq[Expression],
+    keyGroupedPartitioning: Option[Seq[Expression]] = None)
```

Comment:
@liujiayi771 Does this change affect Spark3.2? Because spark 3.2 doesn't contain the `keyGroupedPartitioning `parameter.

Reply:
It will not affect spark3.2, it is an optional parameter. Did you encounter any problems?

Reply:
@liujiayi771 We are currently working on upgrading to Spark 3.4 in Gluten. One of the changes in Spark 3.4 is the modification of BatchScanExec. It introduces a new table parameter. Currently, I have placed all the parameters in the constructor of BatchScanExecTransformer. However the table is not an optional parameter, I am unsure whether this change will affect Spark 3.2 and 3.3 and cause them to encounter this issue.  Do you have any suggestions regarding this? Thank you.

Reply:
@JkSelf According to the description of this PR, I think that such modification in 3.4 will cause the same problems that I encountered before. The reason is as explained in the description. Adding it directly to member variables will cause the lower version spark to not be able to find the corresponding constructors.

Reply:
@JkSelf It may be necessary to re-invent BatchScanTransformer so that it does not inherit Spark's BatchScanExec. Inheriting case class is not a good practice.

Reply:
@liujiayi771 Yes, I will override the `otherCopyArgs ` method in the shim layer to bypass this issue when upgrading to Spark 3.4 [here](https://github.com/oap-project/gluten/pull/3360). We can further optimize the BatchScanExecTransformer in another pull request because it appears that the gluten override the case class for most XXTransformer.